pandas/tools/pivot.py

```diff
@@ -149,9 +149,15 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
         table = table.fillna(value=fill_value, downcast='infer')
 
     if margins:
-        table = _add_margins(table, data, values, rows=index,
-                             cols=columns, aggfunc=aggfunc,
-                             margins_name=margins_name)
+        if dropna:
+            data_dropna = data[data.notnull().all(axis = 1)]
+            table = _add_margins(table, data_dropna, values, rows=index,
```

Comment:
You mean like below? I felt unsafe changing data values but if you're okay with it I'll update the codes.

`        if dropna:
            data = data[data.notnull().all(axis = 1)]
        table = _add_margins(table, data, values, rows=index,
                             cols=columns, aggfunc=aggfunc,
                             margins_name=margins_name)`